default.nix

```diff
@@ -1,66 +1,121 @@
-{ frontend ? false }:
+{ ... } @ args:
+
 let
-  obelisk-src = (import <nixpkgs> {}).fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "obelisk";
-    rev = "6c38599615eddba1b9f8dfb845f7404f53ed8053";
-    sha256 = "0pqhppn2cb69v7r6wbg2zrx5ylxbgq7bl7qilarfbmyd9gcph9h4";
-  };
-  reflex-platform = (import obelisk-src {}).reflex-platform;
 
-  gargoyle-src = reflex-platform.nixpkgs.fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "gargoyle";
-    rev = "2c19c569325ad76694526e9b688ccdbf148df980";
-    sha256 = "0257p0qd8xx900ngghkjbmjnvn7pjv05g0jm5kkrm4p6alrlhfyl";
-  };
-  groundhog-src = reflex-platform.nixpkgs.fetchFromGitHub {
-    owner = "obsidiansystems";
-    repo = "groundhog";
-    rev = "c2f18be45e3233f6268c8468eb0732dd6b2e8009";
-    sha256 = "1r9i78bsnm6idbvp87gjklnr10g7c83nsbnrffkyrn1wmd7zzqdn";
+  obelisk = import ./.obelisk/impl args;
+  reflex-platform = obelisk.reflex-platform;
+  pkgs = obelisk.nixpkgs;
+  inherit (pkgs) lib;
+  haskellLib = pkgs.haskell.lib;
+
+  # Some dependency thunks needed
+  repos = {
+
+    # Not sure why this is needed?
+    gargoyle = pkgs.fetchFromGitHub {
+      owner = "obsidiansystems";
+      repo = "gargoyle";
+      rev = "b641902ff1e798e230e5a101fd22ebfbae3c6a08";
+      sha256 = "0pxa2zn1vy4n9cpal1cy1lcpmzgjvrx0ppmjqjs7ykv7z8647f1g";
+    };
+
+    # Not sure why this is needed?
+    groundhog = pkgs.fetchFromGitHub {
+      owner = "obsidiansystems";
+      repo = "groundhog";
+      rev = "febd6c12a676693b1d7339e54a4d107c4a67fcc3";
+      sha256 = "1q05nrqdzh26r17wsd53sdj106dxh3qlg66pqr3jsi8d63iyaq8k";
+    };
+
+    # bytestring-trie in hackage doesn’t support base 4.11+
```

Comment:
I think the fixes were merged to the upstream repo, but not on hackage right? if so lets instead point to the upstream repo

Reply:
I don't think so. There was a PR to upstream but it was never merged:

https://github.com/wrengr/bytestring-trie/pull/3

Reply:
Ah right, that was it. It's fine for now then